File: frontend/src/api/health.ts
import { apiClient } from "./client";

export interface HealthStatus {
  status: string;
  service: string;
  environment: string;
  timestamp: string;
  razorpay_mode: "RAZORPAY_TEST" | "DEMO";
  ai_mode: "OPENAI" | "MOCK_AI";
}

export async function checkHealth(): Promise<HealthStatus> {
  return apiClient<HealthStatus>("/health");
}

File: app/api/routes/health.py
from fastapi import APIRouter
from app.schemas.common import ResponseEnvelope
from app.core.config import settings
from app.utils.helpers import utcnow

router = APIRouter(prefix="/health", tags=["Health"])


@router.get("", response_model=ResponseEnvelope[dict])
def check_health():
    return ResponseEnvelope(
        success=True,
        data={
            "status": "healthy",
            "service": settings.APP_NAME,
            "environment": settings.APP_ENV,
            "timestamp": utcnow().isoformat(),
            "razorpay_mode": "RAZORPAY_TEST" if settings.RAZORPAY_KEY_ID else "DEMO",
            "ai_mode": "OPENAI" if settings.OPENAI_API_KEY else "MOCK_AI",
        },
        message="RecoverAI service is online and operational.",
    )
